### ISSUE TITLE
Update code for Julia v0.4

### ISSUE THREAD

**Issue by jgoldfar** [jgoldfar/Silo.jl#7]
From the Travis CI log:

```
WARNING: deprecated syntax "[a=>b, ...]" at /home/travis/.julia/v0.4/Silo/deps/build.jl:28.
Use "Dict(a=>b, ...)" instead.
WARNING: [a,b] concatenation is deprecated; use [a;b] instead
 in depwarn at ./deprecated.jl:40
 in oldstyle_vcat_warning at ./abstractarray.jl:26
 in vect at abstractarray.jl:29
 in _find_library at /home/travis/.julia/v0.4/BinDeps/src/dependencies.jl:449
 in satisfy! at /home/travis/.julia/v0.4/BinDeps/src/dependencies.jl:735 (repeats 2 times)
 in anonymous at /home/travis/.julia/v0.4/BinDeps/src/dependencies.jl:790
 in include at ./boot.jl:249
 in include_from_node1 at ./loading.jl:128
 in evalfile at loading.jl:168 (repeats 2 times)
 in anonymous at pkg/entry.jl:624
 in cd at ./file.jl:20
 in build! at pkg/entry.jl:623
 in build at pkg/entry.jl:635
 in anonymous at pkg/dir.jl:28
 in cd at file.jl:20
 in cd at pkg/dir.jl:28
 in build at pkg.jl:62
 in process_options at ./client.jl:295
 in _start at ./client.jl:413
```

**Comment by jgoldfar** [comment]
Also,

```
WARNING: zero{T}(::Type{Ptr{T}}) is deprecated, use Ptr{T}(0) instead.
 in depwarn at ./deprecated.jl:40
 in zero at deprecated.jl:29
 in include at ./boot.jl:249
 in include_from_node1 at ./loading.jl:128
 in include at ./boot.jl:249
 in include_from_node1 at loading.jl:128
 in process_options at ./client.jl:321
 in _start at ./client.jl:413
```

**Comment by jgoldfar** [comment]
Closed by #9 (deprecation warning apparently reverted by v0.5)
